dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramCategoryOptionMapping.java

```diff
@@ -0,0 +1,90 @@
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.program;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+import java.io.Serializable;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.hisp.dhis.category.CategoryOption;
+import org.hisp.dhis.common.DxfNamespaces;
+import org.hisp.dhis.common.EmbeddedObject;
+
+/**
+ * Defines a filter expression fragment for a {@link CategoryOption} within a {@link
+ * ProgramCategoryMapping} to identify an option in either a category option combination or an
+ * attribute option combination to be returned as a program indicator dimension.
+ *
+ * @author Jim Grace
+ */
+@Builder(toBuilder = true)
+@AllArgsConstructor
+@NoArgsConstructor
+@EqualsAndHashCode
+@JacksonXmlRootElement(localName = "optionMapping", namespace = DxfNamespaces.DXF_2_0)
+public class ProgramCategoryOptionMapping implements EmbeddedObject, Serializable {
+
+  /** The category option UID to be mapped */
+  private String optionId;
+
+  /** The category for this category mapping (from the category option UID) */
+  @Setter @Getter @EqualsAndHashCode.Include private transient CategoryOption option;
+
+  /** The filter for this category option */
+  private String filter;
+
+  // -------------------------------------------------------------------------
+  // Getters and setters
+  // -------------------------------------------------------------------------
```

Comment:
I am not sure if you copied this or if you are the origin of this type of divider comments but I try to delete them when I find them so please don't create new work for me :)

Reply:
Ha! These divider comments already permeated the codebase in 2013 when I joined. :) At first I thought they were a bit clumsy but I used them because that was the DHIS2 coding style. I grew to appreciate them because it made it easy to skim the file and find the logic amidst the boilerplate constructors, setters & getters, equals and hashCode, etc. I had not noticed that you are intentionally deleting them.

I suppose that the more we use lombok for our boilerplate code the less important these dividers become. I still think they could be useful in separating the public methods from the "Supportive" methods.

But that's your call; I've removed all the new ones I had added.